src/components/ViewerMessaging/MessagingDefinition.tsx

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import { MessageSchema } from "./schema";
+import { trimDefinitionsName, getArgumentDefinitionLinkId } from "./utils";
+import MessagingArgument from "./MessagingArgument";
+
+interface MessagingDefinitionProps {
+    definitionName: string;
+    schema: MessageSchema;
+}
+
+export default function MessagingDefinition(props: MessagingDefinitionProps) {
+    const { definitionName, schema } = props;
+
+    const trimmedName = trimDefinitionsName(definitionName);
+    const definition = schema.definitions[trimmedName];
+
+    if (definition.type !== "object") {
+        console.warn(
+            "Tried to render definition for non-object type",
+            trimmedName
+        );
+        return null;
+    }
+
+    const id = getArgumentDefinitionLinkId(trimmedName);
+
+    return (
+        <div className="margin-bottom--lg">
+            <h3>
+                <a id={id} href={`#${id}`}>
```

Comment:
Is it intentional that these are rendered as links to themselves? Seems a bit odd. If it's to facilitate "Copy Link", then maybe as a future improvement we can do something a bit more subtle like render little link buttons or "#" next to them on mouseover?

Reply:
Yeah i was aiming for consistency with the rest of docs where every heading is a hyperlink. The `#` next to it on hover is something fancy they did when statically rendering the markdown content headings. Unfortunately since i'm rendering this dynamically i couldn't quite match it.

Reply:
Scratch that. I was able to figure out how to reference the theme heading properly. Updated in https://github.com/geocortex/dev-center/pull/23/commits/ce106dbf5f1270b106c54afe3834896865581d51.

Reply:
Knowing how to reference the theme heading is sweet! Little dissapointed that also didn't make the headers recognizable by the TOC

Reply:
@Noah-Silvera yeah i dug deep into that. It references a `rightToc` property  which is generated at build time. I didn't see any obvious way to update this dynamically at run time. May be worth a feature request to them as it's a pretty interesting use case.